tests/python/pants_test/init/test_plugin_resolver.py

```diff
@@ -115,6 +118,34 @@ def test_exact_requirements(self, unused_test_name, packager_cls):
 
         self.assertEqual(working_set.entries, working_set2.entries)
 
+  @parameterized.expand(INSTALLERS)
+  @skip_unless_python36_and_python37_present
+  def test_exact_requirements_interpreter_change(self, unused_test_name, packager_cls):
+    python36 = PythonInterpreter.from_binary(python_interpreter_path(PY_36))
+    python37 = PythonInterpreter.from_binary(python_interpreter_path(PY_37))
+
+    with self.plugin_resolution(interpreter=python36,
+                                plugins=[('jake', '1.2.3'), ('jane', '3.4.5')],
+                                packager_cls=packager_cls) as results:
+      working_set, chroot, repo_dir, cache_dir = results
+
+      self.assertEqual(2, len(working_set.entries))
+
+      safe_rmtree(repo_dir)
```

Comment:
Brief comment explaining why you do this would help.

```python
# We remove the repo_dir to simulate...
```

Reply:
That's on lines 139-140 below in the `self.fail` message.

Reply:
Agreed. My confusion is why does Py36 still pass? Why do we test Py36 again and delete the repo_dir before doing so?